packages/libs/client/src/signing/keypair/createSignWithKeypair.ts

```diff
@@ -0,0 +1,78 @@
+import { signHash } from '@kadena/cryptography-utils';
+
+import type { ISignFunction, ISingleSignFunction } from '../ISignFunction';
+import { addSignatures } from '../utils/addSignatures';
+
+import type { Debugger } from 'debug';
+import _debug from 'debug';
+import { ICommand, IUnsignedCommand } from '@kadena/types';
+import { parseTransactionCommand } from '../utils/parseTransactionCommand';
+import { IPactCommand } from '../../interfaces/IPactCommand';
+
+const debug: Debugger = _debug('pactjs:signWithKeypair');
+
+export interface IKeypair {
+  publicKey: string;
+  secretKey: string;
+}
+
+/**
+ * Sign with public/private key-pair according to {@link https://github.com/kadena-io/KIPs/blob/master/kip-0015.md | sign-v1 API}
+ *
+ * @public
+ */
+export const createSignWithKeypair: (
+  keyOrKeys: IKeypair | IKeypair[],
+) => ISignFunction & ISingleSignFunction = (
```

Comment:
`ISingleSign` interface already extends `ISingleSignFunction` So you don need to do this again here

Reply:
its better to use function overloads instead
```TS
interface ICreateSignWithKeypair {
    /** jsdoc for this variant */
   ( key: IKeypair) => ISingleSignFunction ,
   
    /** jsdoc for this variant */
   (keys: IKeypair[]) => ISignFunction
}

export const createSignWithKeypair: ICreateSignWithKeypair = ...
```

Reply:
I've modified it to your suggestion